File: solutions/python/saddle-points/1/saddle_points.py
def saddle_points(matrix):
    if any(len(row) != len(matrix[0]) for row in matrix):
        raise ValueError("irregular matrix")
    result = []
    if matrix:
        min_values = [min(row[i] for row in matrix) for i in range(0, len(matrix[0]))]
    for i, row in enumerate(matrix):
        max_val = max(row)
        for j, val in enumerate(row):
            if val == max_val and val == min_values[j]:
                result.append({"row": i + 1, "column": j + 1})
    return result
                

File: solutions/javascript/saddle-points/1/saddle-points.js
//
// This is only a SKELETON file for the 'Saddle Points' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const saddlePoints = (matrix) => {
  const result = [];
  const mins = matrix[0].map((_, ci) => 
    Math.min(...matrix.map(r => r[ci]))
  );
  matrix.forEach((row, ri) => {
    const max = Math.max(...row);
    for(const [ci, value] of row.entries())
      if(value === max && value === mins[ci])
        result.push({row: ri + 1, column: ci + 1});
  });
  return result;
};
